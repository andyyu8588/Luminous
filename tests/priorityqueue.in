import PriorityQueue

function simpleNumberHash(x) {
	return x;	
}

numPriorityQueue = PriorityQueue(simpleNumberHash);

numPriorityQueue.add(1);
numPriorityQueue.add(17);
numPriorityQueue.add(11);
numPriorityQueue.add(0);
numPriorityQueue.add(8);

print(numPriorityQueue.size());
print(numPriorityQueue.deleteMin());
print(numPriorityQueue.deleteMin());
print(numPriorityQueue.deleteMin());
print(numPriorityQueue.deleteMin());
print(numPriorityQueue.deleteMin());
print(numPriorityQueue.isEmpty());


class Object {
	public id;
	public constructor(id) {
		this.id = id;
	}
	public printId() {
		print("Object ID is: " + this.id);
	}
}

function idHash(object) {
	return object.id;
}

objectPriorityQueue = PriorityQueue(idHash);

objectPriorityQueue.add(Object(1));
objectPriorityQueue.add(Object(17));
objectPriorityQueue.add(Object(11));
objectPriorityQueue.add(Object(0));
objectPriorityQueue.add(Object(8));

print(objectPriorityQueue.size());
objectPriorityQueue.min().printId();
objectPriorityQueue.deleteMin().printId();
objectPriorityQueue.deleteMin().printId();
objectPriorityQueue.deleteMin().printId();
objectPriorityQueue.deleteMin().printId();
objectPriorityQueue.min().printId();
objectPriorityQueue.deleteMin().printId();