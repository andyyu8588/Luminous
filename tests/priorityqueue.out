5
0
1
8
11
17
true
5
Object ID is: 0
Object ID is: 0
Object ID is: 1
Object ID is: 8
Object ID is: 11
Object ID is: 17
Object ID is: 17
